pyscitt/pyscitt/client.py

```diff
@@ -15,22 +16,31 @@
 from cryptography.hazmat.primitives.serialization import load_pem_private_key
 from loguru import logger as LOG
 
-from . import crypto
 from .governance import GovernanceClient
 from .prefix_tree import PrefixTreeClient
 from .receipt import Receipt
 
 CCF_TX_ID_HEADER = "x-ms-ccf-transaction-id"
 
+
+class MemberAuthenticationMethod(ABC):
+    @property
+    @abstractmethod
+    def get_key_id(self) -> bytes:
+        pass
+
+    @abstractmethod
+    def sign(self, data: bytes) -> bytes:
+        pass
+
+
 # copied from CCF/tests/infra/clients.py
 class HttpSig(httpx.Auth):
     requires_request_body = True
 
-    def __init__(self, key_id: str, pem_private_key: str):
-        self.key_id = key_id
-        self.private_key = load_pem_private_key(
-            pem_private_key.encode("ascii"), password=None
-        )
+    def __init__(self, member_auth_client: MemberAuthenticationMethod):
+        self.member_auth_client = member_auth_client
+        self.key_id = member_auth_client.get_key_id()
```

Comment:
I don't think it is useful to cache the key id, especially since neither auth client make it expensive to retrieve it. You can just look it up as `self.member_auth_client.key_id` each time `auth_flow` is called. Not a big deal though.